packages/calcite-components/src/components/carousel/carousel.tsx

```diff
@@ -0,0 +1,353 @@
+        this.nextItem();
+        break;
+      case "ArrowLeft":
+        event.preventDefault();
+        this.previousItem();
+        break;
+      case "Home":
+        event.preventDefault();
+        this.setSelectedItem(true, 0);
+        break;
+      case "End":
+        event.preventDefault();
+        this.setSelectedItem(true, this.items?.length - 1);
+        break;
+    }
+  };
+
+  private storeContainerRef = (el: HTMLDivElement): void => {
+    this.container = el;
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+  renderPagination(): VNode {
+    const { activeIndex } = this;
+    const itemGuid = guid();
+
+    return (
+      <div
+        class={{
+          [CSS.pagination]: true,
+          [CSS.isOverlay]: this.controlOverlay,
+        }}
+      >
+        {this.arrowType === "inline" && this.renderPreviousArrow()}
+        {this.items?.map((item, index) => (
+          <Fragment>
+            <calcite-action
+              appearance={this.controlOverlay ? "solid" : "transparent"}
+              class={`pagination-item${index === activeIndex ? " active-icon" : ""}`}
+              icon={index === activeIndex ? ICONS.active : ICONS.inactive}
+              id={`${itemGuid}-${index}`}
+              label={item.label}
+              onClick={() => this.setSelectedItem(true, index)}
+              scale="s"
+              text={item.label}
+            />
+            {!this.tooltipsDisabled && (
+              <calcite-tooltip placement="bottom" reference-element={`${itemGuid}-${index}`}>
+                {item.label}
+              </calcite-tooltip>
+            )}
+          </Fragment>
+        ))}
+        {this.arrowType === "inline" && this.renderNextArrow()}
+      </div>
+    );
+  }
+
+  renderPreviousArrow(): VNode {
+    const dir = getElementDir(this.el);
+    return (
+      <calcite-action
+        appearance={this.controlOverlay ? "solid" : "transparent"}
+        class={CSS.pagePrevious}
+        icon={dir === "rtl" ? ICONS.chevronRight : ICONS.chevronLeft}
+        onClick={this.previousItem}
+        scale={this.arrowType === "edges" ? "m" : "s"}
+        text={this.messages.previous}
+      />
+    );
+  }
+
+  renderNextArrow(): VNode {
+    const dir = getElementDir(this.el);
+    return (
+      <calcite-action
+        appearance={this.controlOverlay ? "solid" : "transparent"}
+        class={CSS.pageNext}
+        icon={dir === "rtl" ? ICONS.chevronLeft : ICONS.chevronRight}
+        onClick={this.nextItem}
+        scale={this.arrowType === "edges" ? "m" : "s"}
+        text={this.messages.next}
+      />
+    );
+  }
+
+  render(): VNode {
+    const { direction, activeIndex } = this;
+    return (
+      <Host>
+        <InteractiveContainer disabled={this.disabled}>
+          <div
+            aria-label={this.label}
```

Comment:
I'm not seeing any role for the carousel. Should this follow the following docs? https://www.w3.org/WAI/ARIA/apg/patterns/carousel/examples/carousel-2-tablist/

Missinge "section" elements, aria-roledescription="carousel", etc. cc @geospatialem

Reply:
From the docs, rotating is optional.

<img width="953" alt="image" src="https://github.com/Esri/calcite-design-system/assets/1231455/36b93360-d247-47b9-aa88-f34883c7cdd6">

Reply:
Its still missing the `aria-roledescription="carousel"` and `section` elements.

Reply:
Ah, _believe_ the reason AT isn't getting on-demand live updates when the `action` is selected is there should also be a unique `aria-controls` value linking to the `carousel-item`. 

For instance:

```html
<calcite-action aria-controls="unique-carousel-id-123xyz" />
<calcite-carousel-item id="unique-carousel-id-123xyz" />
```